Content.Server/Explosion/EntitySystems/RepulseAttractOnTriggerSystem.cs

```diff
@@ -0,0 +1,27 @@
+using Content.Server.Explosion.Components.OnTrigger;
+using Content.Shared.RepulseAttract;
+using Content.Shared.Timing;
+
+namespace Content.Server.Explosion.EntitySystems;
+
+public sealed class RepulseAttractOnTriggerSystem : EntitySystem
+{
+    [Dependency] private readonly RepulseAttractSystem _repulse = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly UseDelaySystem _delay = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<RepulseAttractOnTriggerComponent, TriggerEvent>(OnTrigger);
+    }
+
+    private void OnTrigger(Entity<RepulseAttractOnTriggerComponent> ent, ref TriggerEvent args)
+    {
+        if (TryComp<UseDelayComponent>(ent, out var useDelay)
+            && _delay.IsDelayed((ent, useDelay)))
```

Comment:
```suggestion
        if (_delay.IsDelayed(ent))
```
Just use the overload that gets the component for you.
In case it does not have the component it will return false.

Reply:
youre right, didnt realize that